src/foamadapter/framework/context.py

```diff
@@ -0,0 +1,24 @@
+from typing import Annotated, Any, TypeVar
+
+from pydantic import BaseModel
+
+T = TypeVar("T")
+Model = Annotated[T, "models"]
+Field = Annotated[T, "fields"]
+
+
+class FieldUpdates(dict):
+    """
+    A dictionary-like object that signals to the @step decorator
+    that its contents should be used to update context.fields.
+    """
+
+    pass
+
+
+class Context(BaseModel):
```

Comment:
Same as above Context seams to be an important concept. Can you add a docstring explaining what it represents?